src/test/java/com/google/cloud/tools/libraries/LibrariesTest.java

```diff
@@ -43,20 +62,66 @@ public void testJson() throws FileNotFoundException, URISyntaxException {
     JsonReader reader = factory.createReader(in); 
     Iterator<JsonValue> apis = reader.readArray().iterator();
     Assert.assertTrue(apis.hasNext());
-    for (JsonObject api = (JsonObject) apis.next(); apis.hasNext(); api = (JsonObject) apis.next()) {
+    while (apis.hasNext()) { 
+      JsonObject api = (JsonObject) apis.next();
+      verifyApi(api);
+    }
+  }
+
+  private static final String[] statuses = {"alpha", "beta", "GA"};
+
+  private static void verifyApi(JsonObject api) throws URISyntaxException {
+    Assert.assertFalse(api.getString("name").isEmpty());
+    Assert.assertFalse(api.getString("description").isEmpty());
+    String transport = api.getString("transport");
+    Assert.assertTrue("http".equals(transport) || "grpc".equals(transport));
+    new URI(api.getString("documentation"));
+    if (api.getString("icon") != null) {
+      new URI(api.getString("icon"));
+    }
+    JsonArray clients = api.getJsonArray("clients");
+    Assert.assertFalse(clients.isEmpty());
+    for (int i = 0; i < clients.size(); i++) {
+      JsonObject client = (JsonObject) clients.get(i);
+      String status = client.getString("status");
+      
+      Assert.assertThat(statuses, hasItemInArray(status));
+      new URI(client.getString("apireference"));
+      new URI(client.getString("site"));
+      Assert.assertTrue(client.getString("languageLevel").matches("1\\.\\d+\\.\\d+"));
+      String versionString = client.getString("version");
+      Assert.assertTrue(versionString.matches("\\d+\\.\\d+\\.\\d+"));
+      Assert.assertNotNull(client.getJsonObject("mavenCoordinates"));
+      if (client.getString("source") != null) {
+        new URI(client.getString("source"));
+      }
+    }
+  }
+  
+  @Test
+  public void testMavenCoordinates() throws FileNotFoundException, URISyntaxException {
```

Comment:
could this be named testMavenCoordinate_fileIsValid or something? It's hard for me to tell (since I didn't really look at the previous CL) what this test is doing.